src/api/local/deployTools.ts

```diff
@@ -72,9 +72,24 @@ export namespace DeployTools {
 
           methods.push({ method: "all" as DeploymentMethod, label: `All`, description: `Every file in the local workspace` });
 
-          method = (await vscode.window.showQuickPick(methods,
-            { placeHolder: `Select deployment method to ${remotePath}` }
-          ))?.method;
+          let defaultDeploymentMethod = instance.getConfig()?.defaultDeploymentMethod as DeploymentMethod
+
+          if (methods.find((element) => element.method === defaultDeploymentMethod)) { // default deploy method is usable
+            method = defaultDeploymentMethod
+          
+          } else if (defaultDeploymentMethod as string !== '' && defaultDeploymentMethod !== undefined) { // default deploy method is set but not usable.
```

Comment:
You can you reduce your test to this since you're testing if `defaultDeploymentMethod` is not `falsy`.
```suggestion
          } else if (defaultDeploymentMethod) { // default deploy method is set but not usable.
```